force_wfmanager/ui/setup/mco/kpi_specification_view.py

```diff
@@ -0,0 +1,308 @@
+from traits.api import (
+    Bool, Event, Instance, List, Property, Unicode, cached_property,
+    on_trait_change, HasTraits, Button
+)
+from traitsui.api import (
+    Item, View, ListEditor, TableEditor, ObjectColumn,
+    VGroup, HGroup, UReadonly, ModelView
+)
+
+from force_bdss.api import KPISpecification, BaseMCOModel
+from force_wfmanager.utils.variable_names_registry import (
+    VariableNamesRegistry
+)
+
+
+class TableRow(HasTraits):
+    """A representation of a variable in the workflow. Instances of TableRow
+    are displayed in a table using the TableEditor."""
+
+    #: The variable's type
+    type = Unicode()
+
+    #: The variable's name
+    name = Unicode()
+
+
+class KPISpecificationModelView(ModelView):
+
+    # -------------------
+    # Required Attributes
+    # -------------------
+
+    #: KPI model
+    model = Instance(KPISpecification)
+
+    # ------------------
+    # Regular Attributes
+    # ------------------
+
+    #: Defines if the KPI is valid or not. Set by the function
+    #: map_verify_workflow in workflow_view.py
+    valid = Bool(True)
+
+    #: An error message for issues in this modelview. Updated by
+    #: :func:`verify_tree
+    #: <force_wfmanager.ui.setup.workflow_tree.WorkflowTree.verify_tree>`
+    error_message = Unicode()
+
+    #: Event to request a verification check on the workflow
+    #: Listens to: :attr:`model.name <model>` and :attr:`model.type <model>`
+    verify_workflow_event = Event
+
+    # ------------------
+    #     Properties
+    # ------------------
+
+    #: Human readable label for ModelView
+    label = Property(Instance(Unicode),
+                     depends_on='model:[name,objective]')
+
+    # ------------------
+    #     View
+    # ------------------
+
+    # The traits_view only displays possible options for
+    # model.name listed in kpi_names. However, it is possible
+    # to directly change model.name without updating kpi_names
+    traits_view = View(
+        UReadonly('model.name'),
+        Item("model.objective"),
+        Item('model.auto_scale'),
+        Item("model.scale_factor",
+             visible_when='not model.auto_scale'),
+        kind="subpanel",
+    )
+
+    #: Property getters
+    def _get_label(self):
+        """Gets the label from the model object"""
+        if self.model.name == '':
+            return "KPI"
+        return "KPI: {} ({})".format(self.model.name, self.model.objective)
+
+    #: Listeners
+    @on_trait_change('model.[name,type]')
+    def kpi_change(self):
+        self.verify_workflow_event = True
+
+
+class KPISpecificationView(HasTraits):
```

Comment:
Considering renaming this `KPIView`

Reply:
Ah, this file contains both `KPISpecificationView` and `KPISpecificationModelView` classes, which is confusing I agree. In #313 this has been inadvertently resolved (along with the MCO parameters).